library/blob.c

```diff
@@ -330,10 +330,12 @@ inline static uint64_t eblob_validate_ctl_flags(struct eblob_backend *b, uint64_
 	if (b->cfg.blob_flags & EBLOB_NO_FOOTER)
 		flags |= BLOB_DISK_CTL_NOCSUM;
 
-	if (!(flags & BLOB_DISK_CTL_NOCSUM))
-		flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
-	else
-		flags &= ~BLOB_DISK_CTL_CHUNKED_CSUM;
+	/*
+	 * We have to add BLOB_DISK_CTL_CHUNKED_CSUM because it shows
+	 * that footer was prepared with specific size, even if the record
+	 * has BLOB_DISK_CTL_NOCSUM flag.
+	 */
+	flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
```

Comment:
>I meant different thing.
Let's suppose we have a record with BLOB_DISK_CTL_CHUNKED_CSUM bit set, it also has a footer, but it is empty.
Previously it was enough just to look at record's flag to determine that there is no checksum and thus eblob should not try to verify it.

It continues to be true with one change: it is enough to look at record's flag and if `BLOB_DISK_CTL_NOCSUM` is set then checksum is empty and it shouldn't be verified.

>What's wrong with not setting BLOB_DISK_CTL_CHUNKED_CSUM bit if we do not actually plan to put checksum into footer?

If eblob wasn't configured with `EBLOB_NO_FOOTER` then all records have footer. Footer size is different for records with old and with new checksums. Without `BLOB_DISK_CTL_CHUNKED_CSUM` we can't determine which checksums (new or old) are contained in the footer, so we can't calculate its size.

The presence or absence of `BLOB_DISK_CTL_CHUNKED_CSUM` says nothing about that checksums are filled or not, so it shouldn't be used for such purpose.